src/InvestJob.js

```diff
@@ -0,0 +1,119 @@
+import crypto from 'crypto';
+import localforage from 'localforage';
+
+import { getLogger } from './logger';
+
+const logger = getLogger(__filename.split('/').slice(-1)[0]);
+
+const KEYS_ARRAY = 'sortedHashArray';
+
+/**
+ * Create an object to hold properties associated with an Invest Job.
+ *
+ */
+export default class InvestJob {
+  /* If none exists, init an empty array for the sorted workspace hashes */
+  static async initDB() {
+    const keys = await localforage.getItem(KEYS_ARRAY);
+    if (!keys) {
+      localforage.setItem(KEYS_ARRAY, []);
+    }
+  }
+
+  /* Return an array of job metadata objects, ordered by most recently saved */
+  static async getJobStore() {
+    let jobArray = [];
+    const sortedKeys = await localforage.getItem(KEYS_ARRAY);
+    if (sortedKeys) {
+      jobArray = await Promise.all(sortedKeys.map(
+        (key) => localforage.getItem(key)
+      ));
+    }
+    return jobArray;
+  }
+
+  static async clearStore() {
+    await localforage.clear();
+  }
+
+  /**
+   * @param {object} obj - the metadata property
+   * @param {string} obj.modelRunName - name to be passed to `invest run`
+   * @param {string} obj.modelHumanName - colloquial name of the invest model
+   * @param {object} obj.argsValues - an invest "args dict" with initial values
+   * @param {object} obj.workspace - defines the invest workspace and suffix
+   * @param {string} obj.workspace.directory - path to invest model workspace
+   * @param {string} obj.workspace.suffix - invest model results suffix
```

Comment:
Is it useful to have `suffix` nested under a `workspace` heirarchy? Any reason for not going with `obj.suffix` and `obj.workspace`?

Reply:
Not a very important reason, no. But my reasoning was that it's really the directory & the suffix that together uniquely identify an invest "workspace". So that's why I had them bundled in one object.

At this point it may not even be useful to have them stored anywhere besides inside of `argsValues`. I think I should re-visit that before we merge because storing the same data in multiple places, without a good reason, seems like a bad idea.